datastore/warehouse/structs.go

```diff
@@ -0,0 +1,53 @@
+package warehouse
+
+import (
+	"errors"
+	"os"
+	"time"
+)
+
+var (
+	ErrUnknownType = errors.New("unknown data type")
+)
+
+type DataType uint8
+
+const (
+	GetPayloadRequest DataType = iota
+	SubmitBlockRequest
+)
+
+func toString(data DataType) string {
+	switch data {
+	case GetPayloadRequest:
+		return "GetPayloadRequest"
+	case SubmitBlockRequest:
+		return "SubmitBlockRequest"
+	default:
+		return "unknown"
+	}
+}
+
+type StoreRequest struct {
+	DataType  DataType
+	Data      []byte
+	Slot      uint64
+	Id        string
+	Timestamp time.Time
+
+	header []byte
```

Comment:
Doesn't look used :)

Reply:
It's used in `writeToFile`

Reply:
Can we make it Public / Upercase then so it reads better? :)

Reply:
We could, but I do not recommend it, because it should not be used by the relay component. Header is used for the internal optimization of warehouse.

Reply:
👍